Xamarin.Forms.Core/BindableObject.cs

```diff
@@ -434,9 +434,8 @@ internal void SetValueCore(BindableProperty property, object value, SetValueFlag
 
 		internal void ApplyBindings(bool skipBindingContext, bool fromBindingContextChanged)
 		{
-			var prop = _properties.ToArray();
-			for (int i = 0, propLength = prop.Length; i < propLength; i++) {
-				BindablePropertyContext context = prop [i];
+			for (int i = 0, propLength = _properties.Count; i < propLength; i++) {
+				BindablePropertyContext context = _properties [i];
```

Comment:
there _was_ a reason for this fix to exists. I wish a test for this exists, and will fail 🤞

Reply:
Here's where the ToArray was introduced: https://github.com/xamarin/Duplo/commit/08eae560cc#diff-2e3c6aa6af208011b62f789f3b1300f5L309

Reply:
Huh. Well. I guess we'll find out if it was a _good_ reason or a _bad_ reason soon enough... We could capture the array length and throw if it changes during the loop. That might flush out the issue during testing...